### ISSUE TITLE
duplicate bisect of "git lfs pull origin" hangs indefinitely after upgrade to 2.6.0

### ISSUE THREAD

**Issue by whitty** [git-lfs/git-lfs#3366]
Hi,

This ticket is a duplicate of #3358 - I'm seeing the same problem and was able to run the git-bisect requested - however I didn't want to spam the ticket in case my issue was subtly different from original poster's issue.

For us the issue seems to be repository (ie data) dependent (a single repository on a server with about 10 active LFS repos shows errors only on a single repository) - is that consistent with what you have observed?

Here's the criteria I used: on success pull is instant - this times out after 20 seconds:

```
timeout 20 env GIT_TRACE=1  git -C /path/to/repo lfs pull
```

I ran bisect between v2.6.0 v2.5.0 . Here's git bisect's output

```
e3fcde746a04c26f33ac866f39d54b7cbcf2d933 is the first bad commit
commit e3fcde746a04c26f33ac866f39d54b7cbcf2d933
Author: brian m. carlson <bk2204@github.com>
Date:   Wed Sep 5 14:12:53 2018 +0000

    git: replace object scanner with one based on gitobj
    
    Currently, our object scanner is based on calling git cat-file --batch.
    This is less efficient than reading the objects ourselves in process, so
    update the object scanner to use our gitobj package instead.
    
    Remove the reset function from the object scanner, since it is no longer
    used.  Also, since we no longer need to worry about reading too much
    data from git cat-file --batch, switch the io.LimitedReader to a plain
    io.Reader to simplify the code.
    
    Note that in the tests we now specify the object ID of the pointer item
    to scan since we no longer can implicitly read from a fake buffer.
    Consequently, we no longer look for a trailing entry, since that won't
    succeed anymore.

:040000 040000 bf99741ce3e47b8a63bac52c1b1ad4e57492ba05 f35bb463044a99f82ddac340365606150bdc64af M	git
:040000 040000 869ba607aa2919a7c80a790c371dc10e84277ad7 cf6f5e7a50c3e6c6c97febea0f8483b97d0b37c6 M	lfs
bisect run success
```

I assume you'll need more information than that to diagnose the problem.  Let me know if I can help

**Comment by bk2204** [comment]
Thanks.  This is really helpful information.  Let me see if I can come up with a branch that I think may fix this issue.

**Comment by whitty** [comment]
Closed as duplicate of #3358
